docs: update rbac tier information (#71865)
grafana/grafana: docs/sources/administration/roles-and-permissions/access-control/_index.md

### Basic roles

> Basic roles are the standard roles that are available in Grafana OSS. If you have purchased a Grafana Enterprise license, or Grafana Cloud subscription you can still use basic roles.

Reviewer: ```suggestion
Basic roles are the standard roles that are available in Grafana OSS, and they are still available in every edition of Grafana, like Cloud and Enterprise.
```

### Custom roles

> Create custom roles to manage user permissions in a way that meets your security requirements.

Reviewer: ```suggestion
If you use Grafana Cloud or Enterprise, you can create custom roles to manage user permissions in a way that meets your security requirements.
```
